cpp/include/cudf/io/parquet.hpp

```diff
@@ -195,6 +195,10 @@ class parquet_reader_options {
   /**
    * @brief Sets AST based filter for predicate pushdown.
    *
+   * The filter can utilize cudf::ast::column_name_reference to reference a column by its name,
+   * even if it's not necessarily present in the requested projected columns.
+   * To refer to output column indices, you can use cudf::ast::column_reference.
```

Comment:
*suggestion*: I think this is not fully clear. To my understanding there are two possible column mappings available:

1. The on disk parquet data, this has columns `["A", ..., "X"]` (say), in an order defined by the parquet file
2. The requested projected columns, `["X", "A", "Y"]` (say), in an order defined by caller

If I refer to column indices `[0, 2]` in my filter expression, will these apply to `"X"` and `"Y"` both when being applied to the projected column data (as a post-filter) and the row-group pre-filtering?